### PR TITLE
firecracker-experimental: Fix indentation issues

### PR DESCRIPTION
**Pull request by mcastelino**
Description of changes: Due to indentation issues swagger was unable to generate code.
Fix the indentation issues with vsock fields.

The error was

```
 swagger generate client -f ./firecracker-experimental.yaml --model-package=client/models --client-package=client
2019/06/18 17:20:06 validating spec ./firecracker-experimental.yaml
The swagger spec at "./firecracker-experimental.yaml" is invalid against swagger specification 2.0. see errors :
- definitions.TokenBucket.Vsock in body is a forbidden property
- paths./network-interfaces/{iface_id}./vsocks/{id} in body is a forbidden property
```

Signed-off-by: Manohar Castelino <manohar.r.castelino@intel.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mcastelino** [comment]
Looks like something changed in 0.17.0 due to which the container launches but hangs. debugging it now.

**Comment by andreeaflorescu** [comment]
@mcastelino what tool are you using for testing the swagger file? We also have some [tests for yamls](https://github.com/firecracker-microvm/firecracker/blob/7267a7d73875da3e30bf0901448c56182ba87c20/tests/integration_tests/build/test_style.py#L124) but apparently they're not doing what they're supposed to do. We should add the tool you are using to our integration test system to make sure we don't break it in the future again.

**Comment by mcastelino** [comment]
> @mcastelino what tool are you using for testing the swagger file? We also have some [tests for yamls](https://github.com/firecracker-microvm/firecracker/blob/7267a7d73875da3e30bf0901448c56182ba87c20/tests/integration_tests/build/test_style.py#L124) but apparently they're not doing what they're supposed to do. We should add the tool you are using to our integration test system to make sure we don't break it in the future again.

@andreeaflorescu I use https://github.com/go-swagger/go-swagger but I am open to use the same tool that you use. This seems to match what the firecracker-go-sdk seems to use.

/cc @nmeyerhans

**Comment by andreeaflorescu** [comment]
@mcastelino we should add that tool to our dev container so we can use it in our CI.
We can do two things depending on how urgent is the fix for you:
- merge this PR without the test and do a patch release on top of 0.17.0 to fix the swagger
- add the swagger generator tool to our dev container and also write the integration test before merging this PR.

I prefer the second option but it might take some more time. Is this issue a blocker for you?

**Comment by mcastelino** [comment]
> I prefer the second option but it might take some more time. Is this issue a blocker for you?

@andreeaflorescu the second approach is better. This is not a blocker for us. We will carry the modified file in our repo till the upstream file is fixed.

https://github.com/kata-containers/runtime/pull/1813

**Comment by nmeyerhans** [comment]
We use the [quay.io goswagger containers](https://quay.io/repository/goswagger/swagger?tab=tags) for code generation in the firecracker-go-sdk. There have been a few regressions that we've caught and fixed in the past as well, and you'll find some references to that in the swagger.yaml file's history.

Adding a proper swagger client generation step to the firecracker CI system would be a great guard against future occurrences.

**Comment by andreeaflorescu** [comment]
@mcastelino we tried to find a tool that can be easy to use in Python, but failed. I think for now it's better to merge the fix as is. Since Kata Containers CI runs on Firecracker PRs now, we should not have this problem anymore in the future after you switch to consuming the swagger definition from our repository.

File: api_server/swagger/firecracker-experimental.yaml [hunk 1]
```diff
@@ -354,38 +354,38 @@ paths:
...
-            description: Vsock updated
-          400:
-            description: Vsock cannot be created due to bad input
-            schema:
-              $ref: "#/definitions/Error"
-          default:
-            description: Internal server error
-            schema:
-              $ref: "#/definitions/Error"
+  /vsocks/{id}:
```
Comment (andreeaflorescu) [inline api_server/swagger/firecracker-experimental.yaml]: Please also add an entry about this fix to the CHANGELOG.md file.

**Comment by mcastelino** [comment]
> @mcastelino we tried to find a tool that can be easy to use in Python, but failed. I think for now it's better to merge the fix as is. Since Kata Containers CI runs on Firecracker PRs now, we should not have this problem anymore in the future after you switch to consuming the swagger definition from our repository.

Sure. I can add a job to the Kata CI that validates the experimental and production yamls. 

Today I do one time auto generation and check in the generated files each time we see a firecracker release.

**Comment by andreeaflorescu** [comment]
@mcastelino any update on this PR? To merge it we also need the entry in CHANGELOG.md under the Fixed section.

**Comment by mcastelino** [comment]
> @mcastelino any update on this PR? To merge it we also need the entry in CHANGELOG.md under the Fixed section.

Will update PR today.

**Comment by mcastelino** [comment]
> @mcastelino any update on this PR? To merge it we also need the entry in CHANGELOG.md under the Fixed section.

@andreeaflorescu under which section should I add the fixed entry. 0.17.0 has already been released. So should it be added at the top? `CHANGELOG.md`

**Comment by andreeaflorescu** [comment]
@mcastelino we usually add an Unreleased section at the top of the Changelog. When we release a new Firecracker version, we replace Unreleased with the release number. Here is an example: 0efb4aa8f4afb2879e2f09b9d751635c6e4411ba

File: api_server/swagger/firecracker-experimental.yaml (hunk 1 above)
Comment (mcastelino) [reply api_server/swagger/firecracker-experimental.yaml]: Done

**Pull request merged**